apps/site/pages/en/learn/modules/package-configuration.mdx

```diff
@@ -0,0 +1,597 @@
+---
+title: Package Configuration
+layout: learn
+authors: JakobJingleheimer
+---
+
+# Package configuration
+
+All the provided `package.json` configurations (not specifically marked “does not work”) work in Node.js 12.22.x (v12 latest, the oldest supported line) and 17.2.0 (current latest at the time)[^1], and for grins, with webpack 5.53.0 and 5.63.0 respectively. These are available: [nodejs/package-examples](https://github.com/nodejs/package-examples/blob/main/config).
+
+For curious cats, [How did we get here](#how-did-we-get-here) and [Down the rabbit-hole](#down-the-rabbithole) provide background and deeper explanations.
+
+## Pick your fix
+
+There are 2 main options, which cover almost all use-cases:
+
+- Write source code and publish in CJS (you use `require()`); CJS is consumable by both CJS and ESM (in all versions of node). Skip to [CJS source and distribution](#cjs-source-and-distribution).
+- Write source code and publish in ESM (you use `import`, and don't use top-level `await`); ESM is consumable by both ESM and CJS (in node 22.x and 23.x; see [`require()` an ES Module](https://nodejs.org/api/modules.html#loading-ecmascript-modules-using-require)). Skip to [ESM source and distribution](#esm-source-and-distribution).
+
+It's generally best to publish only 1 format, either CJS _or_ ESM. Not both. Publishing multiple formats can result in the [dual-package hazard](#the-dual-package-hazard), as well as other drawbacks.
+
+There are other options available, mostly for historical purposes.
+
+<table>
+  <thead>
+    <tr>
+      <th>You as a package author write</th>
+      <th>Consumers of your package write their code in</th>
+      <th>Your options</th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td rowSpan="2">CJS source code using <code>require()</code></td>
+      <td>ESM: consumers <code>import</code> your package</td>
+      <td><a href="#cjs-source-and-only-esm-distribution">CJS source and only ESM distribution</a></td>
+    </tr>
+    <tr>
+      <td>CJS & ESM: consumers either <code>require()</code> or <code>import</code> your package</td>
+      <td><a href="#cjs-source-and-both-cjs-amp-esm-distribution">CJS source and both CJS & ESM distribution</a></td>
+    </tr>
+    <tr>
+      <td rowSpan="2">ESM source code using <code>import</code></td>
+      <td>CJS: consumers <code>require()</code> your package (and you use top-level <code>await</code>)</td>
+      <td><a href="#esm-source-with-only-cjs-distribution">ESM source with only CJS distribution</a></td>
+    </tr>
+    <tr>
+      <td>CJS & ESM: consumers either <code>require()</code> or <code>import</code> your package</td>
+      <td><a href="#esm-source-and-both-cjs-amp-esm-distribution">ESM source and both CJS & ESM distribution</a></td>
+    </tr>
+  </tbody>
+</table>
+
+### General notes
```

Comment:
Id move the general notes section at the end since it breaks the reading flow. you just explained there are 2 ways of publishing, lets dive into them and add considerations at the end. Also it requires to know some concepts that are explained after.